### PR TITLE
JAVA-2630: Correctly handle custom classes in IndexMetadata.describe

### PR DESCRIPTION
**Pull request by emerkle826**

### CODE REVIEW

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/IndexMetadataTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Test;
+
+public class IndexMetadataTest {
+
+  private IndexMetadata indexMetadata;
+
+  @Before
+  public void before() {
+    final Map<String, String> options = new HashMap<>();
+    options.put("class_name", "com.datastax.MyClass");
```
Comment (tomekl007) [inline core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/IndexMetadataTest.java]: what is a reason for doing this in `@Before?` From what I see this is used in only one `@Test` so I guess it belongs to that test (given section)

Comment (emerkle826) [reply core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/IndexMetadataTest.java]: Good point. I started it in a `@Before` method as I thought I was going to have to mock things. If we decide to add more tests here, maybe there will be a more obvious reason for a `@Before` method, but for now I can just move it all inside the one test method.

**Review by olim7t — APPROVED** [review]
👍

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/IndexMetadataTest.java (hunk 1 above)
Comment (olim7t) [reply core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/IndexMetadataTest.java]: 👍 to move it to the test method, it's a bit overkill right now.

**Pull request merged**